apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -231,9 +239,11 @@ export default function ListBox({
     // If values have been filtered in the currently loaded page, we want to
     // prevent rendering empty rows by assigning the actual number of items to render
     // since count (qcy) does not reflect this in DQ mode currently.
-    const hasFilteredValues = ps.some((page) => page.qArea.qHeight < MINIMUM_BATCH_SIZE);
+    // If any qMatrix was not 100 length && result of it was less that qcy
+    // then => qTop + qHeight might indicate length of items as long as you scroll and fetch more items
+    const notEnoughItems = ps.some((p) => p.qArea.qHeight !== MINIMUM_BATCH_SIZE);
     const h = Math.max(...ps.map((page) => page.qArea.qTop + page.qArea.qHeight));
```

Comment:
What does qTop in this case mean? Is it the top from the list with or without filtered items?

Reply:
`qTop` is the current batch starting point. 
current batch could be filtered or not

Reply:
Yes, but that starting point is that with or without filtered items? 

If I have 2 batches and the first batch (page) only contains 50 items. Is qTop for the second page 101 or 51?

Reply:
"but that starting point is that with or without filtered items?" -> we don't know it could be both
that is why we are trying to find the last page and based on that calculate the height

qTop in first batch could not be 50 items, only last batch could have qTop of any length less than `MINIMUM_BATCH_SIZE` because of pagination 

If we have any data less than `MINIMUM_BATCH_SIZE` => we have an app with less that 100 itesm => we are in first page => it should render all items at once

Reply:
That is not what qTop stands for. qTop should be the amount you are away from the top. So for first page it should be 0. For page 20 (based on a size of 100) it should be 2000

Reply:
Been thinking a bit around this. Why would we not always return h. That should always be correct right?

Reply:
"That is not what qTop stands for. qTop should be the amount you are away from the top" 👇
exactly, how we detect it in code? by start and stop indexes that virtualisation component gives us. 

"Why would we not always return h?" 👇
because we will endup in this situation. 
VirtualList will think that the total list items are h then it will give up for fetching more items!
![Screenshot 2022-07-22 at 08 56 46](https://user-images.githubusercontent.com/29652890/180381756-714447fb-6728-4544-ae45-050670cca01a.png)